lib/remark-lint-thai.js

```diff
@@ -161,6 +168,29 @@ const rules = [
   // Et cetera
   spell(/คอมโพเ(?:น|น้)น(?:ต์|ท์)/g, 'คอมโพเนนต์'),
   spell(/เทคนิก/g, 'เทคนิค'),
+  spell(/เชอปราง(ค์)?/g, 'เฌอปราง'), //Of course
+
+  // ราชบัณฑิตยสถาน. อ่านอย่างไรและเขียนอย่างไร ฉบับราชบัณฑิตยสถาน (แก้ไขเพิ่มเติม) .-- พิมพ์ครั้งที่ 22.-- กรุงเทพฯ : ราชบัณฑิตยสถาน, 2557
+  spell(/กงศุล/g, 'กงสุล'),
+  spell(/กฏหมาย/g, 'กฎหมาย'),
+  spell(/กลบปาก/g, 'กบปาก'),
+  spell(/กบ(?:ฎ|ถ)/g, 'กบฏ'),
+  spell(/กรรมบท/g, 'กรรมบถ'),
+  spell(/กรรมพันธ์/g, 'กรรมพันธุ์'),
+  spell(/กรรมสิทธิ([^์]|$)/g, 'กรรมสิทธิ์'),
```

Comment:
I recommend using a negative look-ahead assertion instead of matching an extra character:

```suggestion
  spell(/กรรมสิทธิ(?!์)/g, 'กรรมสิทธิ์'),
```

This will make the pattern match `กรรมสิทธิ` only when it is not followed by a การันต์.

Reply:
Ok